### PR TITLE
Added support for bulk mode

### PR DESCRIPTION
**Pull request by shouryataneja**
Updates taken from @psquickitjayant

### CODE REVIEW

**Comment by psquickitjayant** [comment]
@indexzero  we fixed all indentation issues and updated test cases. The build is passing for us as we are using our encrypted file and secret travis variables. But it gets failed as we try to open a pull request. Could you please tell us workaround for this? Thanks

File: .travis.yml [hunk 1]
```diff
@@ -10,8 +10,8 @@ before_install:
   - travis_retry npm install --loglevel=http
   - >
     openssl aes-256-cbc \
-      -K $encrypted_4fb8e306b739_key \
-      -iv $encrypted_4fb8e306b739_iv \
```
Comment (indexzero) [inline .travis.yml]: This is why the TravisCI build is failing here. You need to roll this back.

File: test/config.json.enc [hunk 2]
```diff
@@ -1 +1 @@
-N�4�Q��/�#m8ێ�d��CU�l����]����JQ�Y��R�H�����P�[�h�`0�܆��S�_�SzX>0?�q<�Y�+�%?5��t]�H�g��V]5*�$��Z㈮T���׹�Ep<�pf�@��a�?�ߨ��A&.t�޷4�HN+�[�lc�
```
Comment (indexzero) [inline test/config.json.enc]: This is why the TravisCI build is failing here. You need to roll this back.

File: .travis.yml (hunk 1 above)
Comment (psquickitjayant) [inline .travis.yml]: earlier we kept the original keys (used in your travis.yml) and existing config.json.enc, but we were getting the same result. (build failed), then we used our variables and setup a travis for the account and it worked fine. So, we kept our variables. Travis-ci has mentioned the cause of this mismatch as Security restriction : https://docs.travis-ci.com/user/pull-requests#Security-Restrictions-when-testing-Pull-Requests

Comment (psquickitjayant) [inline .travis.yml]: We need config.json since we are sending actual request to Loggly during the tests. Should not we use only dummy values?

Comment (psquickitjayant) [inline .travis.yml]: @indexzero  can we do something here? Thanks!

Comment (indexzero) [inline .travis.yml]: @psquickitjayant could you follow-up with me offline about the account credentials? You work at Loggly right? If Loggly can provide credentials to a free account for this module I think that would fix the problem.

Comment (psquickitjayant) [inline .travis.yml]: Sure

Comment (psquickitjayant) [inline .travis.yml]: @indexzero  can you please confirm, are you talking about Loggly free account? In this case, we  may need a work email address. 

I am suggesting to make mock test cases which will remove the service dependency.

Comment (mostlyjason) [inline .travis.yml]: @indexzero it seems like best practice for unit tests is create mock interfaces for external service dependencies. Is there any reason you coded these to require a call the actual Loggly service? If not, I agree with @psquickitjayant. The test cases will run faster and be a better test of code correctness. Are you ok with this if we converted the test cases over to mocks and submitted an updated PR?

Comment (indexzero) [inline .travis.yml]: Yes if it's an option that can be disabled locally. In other words: I'm fine with Travis hitting mocks, but tests need to be able to hit the live API when debugging locally. 

Would love to see this land. Hopefully that change can push it over the finish line.

Comment (mostlyjason) [inline .travis.yml]: @psquickitjayant ok I guess we can default to mocks so it works in Travis. To run locally against a live API I think we'd need a place for the customer to input their subdomain and token. Can these be variables the tester configures? I'm not sure where they would put them but perhaps in the grunt configuration?

Comment (indexzero) [inline .travis.yml]: Agreed. This was the original approach before I started encrypting the contents. I believe it was `test/config.json`, which is in the `.gitignore`. If it exists, load it. If not and the flag (which should be an environment variable) is set, then throw and exit.

**Comment by shouryataneja** [comment]
Closing this pull request. 
Will be creating new after making mock test cases.

**Pull request closed without merging**